### ISSUE TITLE
Flatpak / Flathub

### ISSUE THREAD

**Issue by Doomsdayrs** [Revolutionary-Games/Thrive-Launcher#158]
**Description**
Consider adding flatpak support and possibly submitting to flathub?

The application is open source after all, and if added to Flathub, it will be super easy to install onto the SteamDeck and other modern Linux desktop distributions.

**Additional context**
https://flathub.org
https://docs.flatpak.org
https://docs.flatpak.org/en/latest/electron.html

**Comment by Doomsdayrs** [comment]
Also, compared to RPM and DEB, you wont have to worry too much regarding distribution specific issues, as the flatpak will have the same environment

**Comment by hhyyrylainen** [comment]
We use https://www.electron.build/ to bundle the launcher into packages. From a quick glance that doesn't support flatpak. So we can't make flatpaks until that tool is updated (or we switch the used launcher technologies).

**Comment by Doomsdayrs** [comment]
@hhyyrylainen In that case, we can just extract the app image

**Comment by hhyyrylainen** [comment]
By "we" who do you mean? If you'd like us to "officially" make those builds, I'd need a PR from someone adding a script to this repo that will do the postprocessing for the appimage on Linux (preferably a ruby script as the existing scripts use that).

**Comment by Doomsdayrs** [comment]
I apologize for the misunderstanding, I use the term "we" on github issues in an inclusive way. Allowing anyone who possibly wants to contribute to feel included with the statement.

----

While flatpak bundles can be produced, they are certain things that need to be setup for such a method of distribution. Its easier to just make the repository, then PR to flathub in my opinion.

This is what a submission to flathub (the repository not the submit PR) looks like: https://github.com/Doomsdayrs/flathub/tree/com.lunarclient.LunarClient

So in this case, when I get home we can easily put together the configuration files to add Thrive to Flathub.

**Comment by Doomsdayrs** [comment]
Done :D
https://github.com/Doomsdayrs/flathub/tree/com.revolutionarygamesstudio.ThriveLauncher

**Comment by Doomsdayrs** [comment]
![Screenshot from 2022-04-16 23-52-21](https://user-images.githubusercontent.com/38189170/163699607-a46d129b-44fc-4ce2-9735-bf669222df6d.png)
![Screenshot from 2022-04-16 23-51-06](https://user-images.githubusercontent.com/38189170/163699608-09061657-946e-46f7-a654-2b9c6b14f8a8.png)

**Comment by Doomsdayrs** [comment]
A few minor bugs to squash, and I can say this is good to add to Flathub

**Comment by Doomsdayrs** [comment]
@hhyyrylainen Would you like to take ownership of the flatpak repo if the PR is merged?

**Comment by hhyyrylainen** [comment]
I'm a bit hesitant about that if it comes with a maintenance burden... but other than that having the repo be under the Revolutionary-Games org should be doable.

**Comment by Doomsdayrs** [comment]
@hhyyrylainen 
Maintenance wise, it would just be Updating the file `com.revolutionarygamesstudio.ThriveLauncher.yaml`

A handy guide that is only 3 steps
Steps to take when updating the launcher:
1. Update sources at line 32 & 33 to the new URL and checksum
2. Update line 47 so the `unappimage` unpacks the new file name
3. Push to repo

Asides, you sparely update the launcher anyhow. So these 3 steps only happen every few months for us. (as I can also help maintain the flatpak)

**Comment by Doomsdayrs** [comment]
Here is some other basic knowledge for you to have https://github.com/flathub/flathub/wiki/App-Maintenance

But I am over here, watching as well ;P , So do remember the burden is shared

**Comment by hhyyrylainen** [comment]
I suppose it'll be easy enough to keep up to date, then (as long as we don't switch the launcher tech stack). So how do we do that repo transfer? Or should I make a new one where you can push your commits with a PR?

**Comment by Doomsdayrs** [comment]
I apologize, it seems their tech stack isn't cut out for working with repos not under the flathub group. (I contacted them on matrix)

**Comment by Doomsdayrs** [comment]
@hhyyrylainen Regarding the flatpak

Should it be named "Thrive", in notation with the game, or "Thrive Launcher", in notation to it being the launcher for the game

Since Minecraft has a launcher, and it is still called Minecraft

But thinking appropriately for the release notes and etc, it would be regarding the launcher (which is being updated via flatpak) and not the game. As the launcher would handle the game updates.

TLDR;
Should the name be `Thrive` or `Thrive Launcher`?

**Comment by hhyyrylainen** [comment]
So that solves the repo problem then...

I think maybe it's best to follow Minecraft's lead here and have the launcher be named "Thrive", as perhaps anyway the patch notes for the game releases won't get updated to flathub with any kind of certainty, but the launcher releases would be (as manual action is needed anyway).

**Comment by Doomsdayrs** [comment]
https://github.com/flathub/com.revolutionarygamesstudio.ThriveLauncher
